container/bin/entrypoint.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh -e
+#
+# SPDX-FileCopyrightText: Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+image_path=$1
+uuid=$(uuidgen)
+socket_path=/tmp/nbd.sock
+ticket_path=ticket.json
+
+gen-ticket.py --ticket-id $uuid -s $socket_path $image_path $ticket_path
+
+nbd-server.py $image_path &
+
+echo "Serving image with ID: $uuid"
+/venv/bin/ovirt-imageio -t $ticket_path &
+
+# Wait for any process to exit
+wait -n
+
+# Exit with status of process that exited first
+exit $?
```

Comment:
Having one python script that does everything will be easier to understand and maintain. Now we have to use a shell script (with all the trouble of shell scripts), think about integrating with 2 additional tools.

Do you expect that the gen-ticket and nbd-server scripts have specific use cases
outside of the container?

Reply:
No, I changed my mind multiple times during the process, on how to structure the script. But I didn't think of specific uses outside the container, I was focused on the container.

Started with python entrypoint, actually. One script for all. But ended up with shell script coming from: https://docs.docker.com/config/containers/multi-service_container/
Also, I've seen this approach used in other repositories.

The main entrypoint script highlights pretty well what are the main services that the container runs in the background, and then waits for them. Makes the purpose clearer that if we do everything in one script. The main problem of it being a shell script in this context is that having more complex arguments is a bit of a pain, but since the idea was to keep the CLI simple, for now it works fine.

With respect to gen-ticket and `nbd-server` being separated, again, I did start them together. Then I felt like they were different things (mostly joint by the socket), and opted to split them, leaving the ticket generation as a helper script, and the `nbd-server` as an additional service run in the container. If we ever move that code to `ovirt-imageioctl`, I guess we would use only the `nbd-server` part anyway. But I'm open to change it.